### PR TITLE
Gh11292 AD_Workflow missing node 

### PR DESCRIPTION
**Pull request by metas-rc**
* Add error message
* set the first node as starting node in workflow
https://github.com/metasfresh/metasfresh/issues/11292

### CODE REVIEW

File: backend/de.metas.material/planning/src/main/java/de/metas/material/planning/pporder/impl/PPRoutingRepository.java [hunk 1]
```diff
@@ -130,7 +135,13 @@ private PPRouting toRouting(final I_AD_Workflow routingRecord)
 				.map(productRecord -> toRoutingProduct(productRecord, nodeIdToActivityIds))
 				.collect(ImmutableList.toImmutableList());
 
-		final PPRoutingActivityId firstActivityId = PPRoutingActivityId.ofAD_WF_Node_ID(routingId, routingRecord.getAD_WF_Node_ID());
+		final int wfNodeId = routingRecord.getAD_WF_Node_ID();
+
+		if(wfNodeId <= 0)
+		{
+			throw new AdempiereException(MSG_AD_Workflow_Missing_Node, routingRecord);
```
Comment (teosarca) [inline backend/de.metas.material/planning/src/main/java/de/metas/material/planning/pporder/impl/PPRoutingRepository.java:142]: instead of routingRecord i think only routing's name would be much more use friendly

File: backend/de.metas.material/planning/src/main/java/de/metas/material/planning/pporder/impl/PPRoutingRepository.java [hunk 2]
```diff
@@ -311,7 +322,7 @@ public void changeRouting(@NonNull final PPRoutingChangeRequest changeRequest)
 
 			for (final I_AD_WF_Node routingActivityRecord : routingActivityRecords)
 			{
-				final PPRoutingActivityId activityId = PPRoutingActivityId.ofAD_WF_Node_ID(routingId, routingActivityRecord.getAD_WF_Node_ID());
+				final PPRoutingActivityId activityId = PPRoutingActivityId.ofAD_WF_Node_ID(routingId, routingRecord.getAD_WF_Node_ID());
```
Comment (teosarca) [inline backend/de.metas.material/planning/src/main/java/de/metas/material/planning/pporder/impl/PPRoutingRepository.java:325]: why this change?
i think `routingActivityRecord` shall be used

File: backend/de.metas.material/planning/src/main/java/de/metas/material/planning/interceptor/AD_WF_Node.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+@Interceptor(I_AD_WF_Node.class)
+@Component
+public class AD_WF_Node
+{
+	final IPPRoutingRepository ppRoutingRepo = Services.get(IPPRoutingRepository.class);
+
+	@ModelChange(timings = ModelValidator.TYPE_AFTER_NEW)
+	public void setStartNodeInWorkflow(final I_AD_WF_Node node)
+	{
+		final PPRoutingId routingId = PPRoutingId.ofRepoIdOrNull(node.getAD_Workflow_ID());
```
Comment (teosarca) [inline backend/de.metas.material/planning/src/main/java/de/metas/material/planning/interceptor/AD_WF_Node.java:44]: isn't `node.getAD_Workflow_ID()` mandatory?

IMHO this check introduces FUD because u cannot get to this point without having AD_Workflow_ID set.

**Review by teosarca — CHANGES_REQUESTED** [review]
check my comments

File: backend/de.metas.material/planning/src/main/java/de/metas/material/planning/pporder/impl/PPRoutingRepository.java (hunk 2 above)
Comment (metas-rc) [reply backend/de.metas.material/planning/src/main/java/de/metas/material/planning/pporder/impl/PPRoutingRepository.java:325]: By mistake. undo-redo then wrong copy paste. And then I didn't notice it. Thanks for find it!

File: backend/de.metas.material/planning/src/main/java/de/metas/material/planning/interceptor/AD_WF_Node.java (hunk 3 above)
Comment (metas-rc) [reply backend/de.metas.material/planning/src/main/java/de/metas/material/planning/interceptor/AD_WF_Node.java:44]: It is. Changing.
